### PR TITLE
Test for accessing single file shared through public link

### PR DESCRIPTION
**Pull request by dpakach**
## Description

Add test for accessing single file shared through public link

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] Backport (if applicable set "backport-request" label and remove when the backport was done)

### CODE REVIEW

File: tests/acceptance/features/lib/PublicLinkFilesPage.php [hunk 1]
```diff
@@ -35,6 +37,8 @@ class PublicLinkFilesPage extends FilesPageBasic {
 	protected $fileListXpath = ".//tbody[@id='fileList']";
 	protected $emptyContentXpath = ".//div[@id='emptycontent']";
 	protected $addToYourOcBtnId = "save-button";
+	protected $singleFileDownloadBtnId = "//a[@id='downloadFile']";
```
Comment (paurakhsharma) [inline tests/acceptance/features/lib/PublicLinkFilesPage.php]: This should be download button Xpath.

File: tests/acceptance/features/webUISharingExternal/shareByPublicLink.feature [hunk 2]
```diff
@@ -131,4 +131,10 @@ So that public sharing is limited according to organization policy
...
 		And the public accesses the last created public link using the webUI
-		Then the file "lorem.txt" should be listed on the webUI
\ No newline at end of file
+		Then the file "lorem.txt" should be listed on the webUI
+
+	Scenario: public should be able to access the shared file through public link
+		When the user creates a new public link for the file 'lorem.txt' using the webUI
+		And the public accesses the last created public link using the webUI
+		Then the text preview of the public link should contain "Lorem ipsum dolor sit amet, consectetur"
+		And the content of the file shared by last public link should be same "lorem.txt"
```
Comment (paurakhsharma) [inline tests/acceptance/features/webUISharingExternal/shareByPublicLink.feature]: s/should be same/should be same as

**Review by paurakhsharma — COMMENTED** [review]
Looks great. Let's see what @phil-davis has to say.

File: tests/acceptance/features/lib/PublicLinkFilesPage.php [hunk 3]
```diff
@@ -152,6 +156,43 @@ public function moveFileTo(
...
+	 *
+	 * @return string|null
+	 * @throws ElementNotFoundException
+	 */
+	public function getDownloadUrl() {
+		$downloadBtn = $this->find("xpath", $this->singleFileDownloadBtnXpath);
+		if ($downloadBtn === null) {
+			throw new ElementNotFoundException(
+				__METHOD__ . " xpath $this->singleFileDownloadBtnXpath ".
+				" could not find preview text container");
```
Comment (phil-davis) [inline tests/acceptance/features/lib/PublicLinkFilesPage.php]: s/could not find preview text container/could not find download button

File: tests/acceptance/features/bootstrap/WebUISharingContext.php [hunk 4]
```diff
@@ -883,6 +884,43 @@ public function createPublicShareLink($name, $settings = null) {
...
+	public function theTextPreviewOfThePublicLinkShouldContain($content) {
+		$previewText = $this->publicLinkFilesPage->getPreviewText();
+		PHPUnit_Framework_Assert::assertContains(
+			$content, $previewText,
+			__METHOD__ . " file preview does not contain expected content"
+		);
+	}
+
+	/**
+	 * @Then the content of the file shared by last public link should be same as :originalFile
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/WebUISharingContext.php]: s/should be same as/should be the same as/

**Review by phil-davis — COMMENTED** [review]
See comments

**Review by phil-davis — APPROVED** [review]

**Comment by individual-it** [comment]
please make a backport

**Pull request merged**